util/gh-pages/script.js

```diff
@@ -0,0 +1,368 @@
+            return {
+                restrict: 'A',
+                link: function (scope, element, attr) {
+                    if (scope.$last === true) {
+                        $timeout(function () {
+                            scope.$emit(attr.onFinishRender);
+                        });
+                    }
+                }
+            };
+        })
+        .controller("lintList", function ($scope, $http, $timeout) {
+            // Level filter
+            var LEVEL_FILTERS_DEFAULT = {allow: true, warn: true, deny: true, none: true};
+            $scope.levels = LEVEL_FILTERS_DEFAULT;
+            $scope.byLevels = function (lint) {
+                return $scope.levels[lint.level];
+            };
+
+            var GROUPS_FILTER_DEFAULT = {
+                cargo: true,
+                complexity: true,
+                correctness: true,
+                deprecated: false,
+                nursery: true,
+                pedantic: true,
+                perf: true,
+                restriction: true,
+                style: true,
+                suspicious: true,
+            };
+            $scope.groups = GROUPS_FILTER_DEFAULT;
+            const THEMES_DEFAULT = {
+                light: "Light",
+                rust: "Rust",
+                coal: "Coal",
+                navy: "Navy",
+                ayu: "Ayu"
+            };
+            $scope.themes = THEMES_DEFAULT;
+
+            $scope.versionFilters = {
+                "≥": {enabled: false, minorVersion: ""},
+                "≤": {enabled: false, minorVersion: ""},
+                "=": {enabled: false, minorVersion: ""},
+            };
+
+            $scope.selectTheme = function (theme) {
+                setTheme(theme, true);
+            }
+
+            $scope.toggleLevels = function (value) {
+                const levels = $scope.levels;
+                for (const key in levels) {
+                    if (levels.hasOwnProperty(key)) {
+                        levels[key] = value;
+                    }
+                }
+            };
+
+            $scope.toggleGroups = function (value) {
+                const groups = $scope.groups;
+                for (const key in groups) {
+                    if (groups.hasOwnProperty(key)) {
+                        groups[key] = value;
+                    }
+                }
+            };
+
+            $scope.selectedValuesCount = function (obj) {
+                return Object.values(obj).filter(x => x).length;
+            }
+
+            $scope.clearVersionFilters = function () {
+                for (let filter in $scope.versionFilters) {
+                    $scope.versionFilters[filter] = { enabled: false, minorVersion: "" };
+                }
+            }
+
+            $scope.versionFilterCount = function(obj) {
+                return Object.values(obj).filter(x => x.enabled).length;
+            }
+
+            $scope.byVersion = function(lint) {
```

Comment:
Other `by<Property>` have a separate function for the filter validation and creation and the application of that filter. Take `byGroup` for instance. This is mainly due to performance, as this will be executed for every lint (~500). I think it would also be better to do the same here. So, I would move the `validateVersionStr` function and everything that enables disables filters somewhere else.

Then we can also simplify the loop to look something like this:

```js
for (const filter in filters) {
    if filters[filter].enabled {
        switch (filter) {
            case "≥":
                if (lintMinorVersion >= filers[filter].minorVersion) return false;
                break;
            case "≤":
                if (lintMinorVersion <= filers[filter].minorVersion) return false;
                break;
            case "=":
                if (lintMinorVersion === filers[filter].minorVersion) return false;
                break;
            default:
                return true
        }
    }
}
```